tests/testsuite/test_acceptance_with_browsers.py

```diff
@@ -121,10 +122,25 @@ def wait_for_reconfigure():
 @pytest.fixture(params=BROWSERS)
 def browser(request):
     browser_name = request.param
-    with splinter.Browser(browser_name, headless=True, wait_time=15, **BROWSER_OPTS[browser_name]) as browser:
-        browser.driver.set_window_size(1920, 1080)
-        browser.visit(URL)
-        yield browser
+    retries = 3
+    while retries > 0:
+        retries -= 1
+        try:
+            with splinter.Browser(
+                        browser_name,
+                        headless=True,
+                        wait_time=15,
+                        **BROWSER_OPTS[browser_name],
+                    ) as browser:
+                browser.driver.set_window_size(1920, 1080)
+                browser.visit(URL)
+                yield browser
+                return
+        except WebDriverException:
```

Comment:
I'd move the `try/except` to only cover the two calls to the browser (`set_window_size` and `visit`). Then move `yield` and `return` to an `else` clause for the `try/except`. The goal is to narrow the `try/except` to only the lines where we expect a failure.

Reply:
`splinter.Browser(...)` is what throws the `WebDriverException`

Reply:
i'll put the `try... except` around the `splinter.Browser(...)` and stop using `with`